defender/poap/autotasks/mint-poap/index.js

```diff
@@ -1,42 +1,92 @@
+const stackName = 'poap_minter';
+const tokenTypeSecretName = `${stackName}_TOKEN_TYPE`;
+const nftIdSecretName = `${stackName}_NFT_ID`;
+const tokenAddressSecretName = `${stackName}_TOKEN_ADDRESS`;
+
 const ethers = require('ethers');
 const { DefenderRelaySigner, DefenderRelayProvider } = require('defender-relay-client/lib/ethers');
-const ERC1155Abi = require('../../../../abi/contracts/ERC1155Mock.sol/ERC1155Mock.json');
-const ERC721Abi = require('../../../../abi/contracts/ERC721Mock.sol/ERC721Mock.json');
-const ERC20Abi = require('../../../../abi/contracts/ERC20Mock.sol/ERC20Mock.json');
-import { TokenTypeEnum, TOKEN_TYPE, ERC1155ID, TOKEN_ADDRESS, domain, types } from '../../constants';
+const ERC1155Abi = ['function mint(address account, uint256 id, uint256 amount, bytes data)'];
+const ERC1155AbiV2 = ['function mint(address to, uint256 id, uint256 amount)'];
+const ERC721Abi = ['function safeMint(address to)'];
+const ERC20Abi = ['function mint(address to, uint256 amount)'];
+
+const domain = { name: 'OpenZeppelin POAP Autotask', version: '1', chainId: '5' };
+const types = {
+  attendee: [
+    { name: 'name', type: 'string' },
+    { name: 'wallet', type: 'address' },
+  ],
+};
 
 exports.handler = async function (event) {
-  if (TOKEN_ADDRESS == ethers.constants.AddressZero) throw new Error('Token address not specified');
+  if (!event) { throw new Error('event undefined'); }
+  const { secrets } = event;
+  if (!secrets) { throw new Error('secrets undefined'); }
+
+  const tokenType = secrets[tokenTypeSecretName];
+  const tokenAddress = secrets[tokenAddressSecretName];
+  const nftQty = ethers.BigNumber.from(1);
+  // ERC1155 Specific data
+  const nftId = ethers.BigNumber.from(secrets[nftIdSecretName]);
+  const nftData = 0x00;
+
+  if (typeof tokenType !== 'string') { throw new Error('Token type not specified') }
+  if (!ethers.utils.getAddress(tokenAddress)) { throw new Error('Valid token address not specified') };
```

Comment:
This does not need the conditional `throw` statement. If the address is not valid, the `ethers` command will throw an Error on its own:
```suggestion
  ethers.utils.getAddress(tokenAddress);
```